lib/client-assets.php

```diff
@@ -66,22 +66,9 @@ function gutenberg_override_script( $scripts, $handle, $src, $deps = array(), $v
 		$script->src  = $src;
 		$script->deps = $deps;
 		$script->ver  = $ver;
-		$script->args = $in_footer;
-
-		/*
-		 * The script's `group` designation is an indication of whether it is
-		 * to be printed in the header or footer. The behavior here defers to
-		 * the arguments as passed. Specifically, group data is not assigned
-		 * for a script unless it is designated to be printed in the footer.
-		 */
-
-		// See: `wp_register_script` .
-		unset( $script->extra['group'] );
-		if ( $in_footer ) {
-			$script->add_data( 'group', 1 );
-		}
+		$script->args = $in_footer ? 1 : null;
 	} else {
-		$scripts->add( $handle, $src, $deps, $ver, $in_footer );
+		$scripts->add( $handle, $src, $deps, $ver, ( $in_footer ? 1 : null ) );
```

Comment:
A bit of a nitpick, but can we use `true`/`false` instead of `1`/`null`? I know that they technically do the same thing, and we wouldn't have any issues using 1/null here... but newer PHP versions are a bit more strict when it comes to types, so just to be safe for the future we should probably stick to boolean values. 👍

Reply:
No, `null` is the default value and only `1` is supported by the [linked special case](https://github.com/WordPress/wordpress-develop/blob/573581f43f63f12cc37620172432cf93745f568c/src/wp-includes/class.wp-scripts.php#L541-L542).

Reply:
Ah you're right... my bad! Thanks for pointing me in the right direction 👍